### DISCUSSION TITLE
W7 desktop client, non-linear zoom-out / zoom-in after startup

### DISCUSSION THREAD

**Discussion by ghost** [standardnotes/forum#2496] [General] [answered]
For Standard Notes Windows 7 desktop client version 1.2.1

The client's zoom-out / zoom-in functions appear to operate non-linearly immediately after starting the client.

1. Start the Windows desktop client, then decide that you'd like the view the notes' fonts at a smaller size.

2. Use either **View** > **Zoom Out** or press CTRL+- to zoom out _one_ level. The client "jumps" the notes' font to _two_ sizes smaller.

3. Use either **View** > **Zoom In** or press CTRL+SHIFT+= to zoom in one level. The desktop client displays the note's font at an intermediate size that's between the initial default size per Step 1 and the two-steps smaller size per Step 2.

4. Close the desktop client, reopen it, and repeat Steps 1 through 3, above. Behavior can be reproduced ever time.

I believe the client performs other non-linear zoom actions. But I won't research those unless necessary...

Cheers & hope this helps,
Riley
SFO

**Comment by atmoio** [comment]
Can you check whether all Electron apps behave this way on your OS? (Like Slack)

**Comment by ghost** [comment]
The Slack client for Windows, *2.6.3 64-bit Direct Download (XO)* doesn't seem to behave the same way.

Here's a composite image containing three screenshots. As captioned, the images show the font size at startup, after clicking **Zoom Out**, and after clicking **Zoom In**. 

![standardnoteszoomoutinout01a](https://user-images.githubusercontent.com/3333327/28679473-2cc06414-72a8-11e7-8447-e2b5efe8c970.png)

All three of the small screenshots were taken without resizing either them or the desktop client to make it easier to see the progression from big > too small > just right.

I'm not familiar with "Electron apps" so the Slack client for Windows is the the only other one I tried.

If no one else seems to be having this problem just set it aside. It's not a big deal for me -- I submitted it more in the spirit of FYI...

Cheers & hope this helps,
Riley
SFO

**Comment by ghost** [comment]
By the by, this problem still exists in the Version 2.0.0 client.

![standardnoteswindowsclientv02zoomproblem](https://user-images.githubusercontent.com/3333327/31623902-d4c3a332-b255-11e7-87c8-5f31d462d824.png)

Similarly, the client doesn't remember the last-used window dimensions. Not sure if all this is because I have a 24-inches monitor attached to a laptop with a smaller screen. My thought being that perhaps the  client is remembering the zoom level / window dimensions for the smaller screen even though the client window is opening on the 24-inches monitor...

Cheers, thanks, & hope this helps,
Riley
SFO

**Comment by atmoio** [comment]
Your app doesn't remember the last used window dimensions? I just tested on Mac and Windows 10 and if I resize the window, quit the app, and re-open it, the dimensions are correctly restored.

**Comment by ghost** [comment]
Correct -- I need to drag the bottom of the client window down every time I start it.

I actually submitted this a separate issue back in July:

https://github.com/standardnotes/forum/issues/30

Perhaps we should discuss that particular issue over there? Central point:

+ It has always been that way.
+ No other applications such as the Slack client Window, demonstrate this behavior.

Neither of these issues are big deals. It'd just be fun if the zoom and window sizing was as elegant as the rest of the program...

Cheers,
Riley

**Comment by myreli** [comment, marked as answer]
Since this first report, the UI has changed considerably, and the new responsive layout handles both zoom and dimensions fine across platforms. 

I will mark it resolved as part of our triage of older issues.

Please let us know a new issue if you face any problems. Thanks!
